frontend/src/metabase/core/components/EditableText/EditableText.styled.tsx

```diff
@@ -27,12 +27,16 @@ export const EditableTextRoot = styled.div<EditableTextRootProps>`
       border-color: ${color("border")};
     `}
 
-  &:after {
-    content: attr(data-value);
-    visibility: hidden;
-    white-space: pre-wrap;
-    word-wrap: break-word;
-  }
+  ${({ isEditingMarkdown }) =>
+    isEditingMarkdown &&
+    css`
+      &:after {
+        content: attr(data-value);
+        visibility: hidden;
+        white-space: pre-wrap;
+        word-wrap: break-word;
```

Comment:
I think we should add `word-wrap: break-word` to the container too. Otherwise, we will still experience the horizontal overflow.

This screenshot is from running this PR
![image](https://github.com/metabase/metabase/assets/1937582/e370ef43-c5bb-443c-aba8-c52dfd2d9025)